mycroft/skills/diagnostics/__init__.py

```diff
@@ -0,0 +1,165 @@
+        self.register_intent(cpu_intent, self.handle_cpu_intent)
+
+        drive_intent = IntentBuilder("DriveIntent")\
+            .require("DriveSpaceKeyword")\
+            .build()
+        self.register_intent(drive_intent, self.handle_drive_intent)
+
+        # There's already an IP skill.
+        # ip_intent = IntentBuilder("IpIntent")\
+        #    .require("IpKeyword")\
+        #    .build()
+        # self.register_intent(ip_intent, self.handle_ip_intent)
+
+        uptime_intent = IntentBuilder("UptimeIntent")\
+            .require("UptimeKeyword")\
+            .build()
+        self.register_intent(uptime_intent, self.handle_updtime_intent)
+
+        custom_intent = IntentBuilder("CustomIntent")\
+            .require("CustomKeyword")\
+            .build()
+        self.register_intent(custom_intent, self.handle_custom_intent)
+
+    def handle_cpu_intent(self, message):
+        data = {
+            "percent": psutil.cpu_percent(interval=1)
+        }
+        self.speak_dialog("cpu", data)
+
+    def handle_drive_intent(self, message):
+        partitions = psutil.disk_partitions()
+        for partition in partitions:
+            partition_data = psutil.disk_usage(partition.mountpoint)
+            # total=21378641920, used=4809781248, free=15482871808,
+            # percent=22.5
+            data = {
+                "mountpoint": partition.mountpoint,
+                "total": sizeof_fmt(partition_data.total),
+                "used": sizeof_fmt(partition_data.used),
+                "free": sizeof_fmt(partition_data.free),
+                "percent": partition_data.percent
+            }
+            if partition_data.percent >= 90:
+                self.speak_dialog("drive.low", data)
+            else:
+                self.speak_dialog("drive", data)
+
+    def handle_ip_intent(self, message):
+        ips = subprocess.check_output(['hostname', "-I"])
+        ips = ips.strip()
+        ips = ips.split(" ")
+
+        public_json = subprocess.check_output([
+            "wget", "-qO-",
+            "https://api.ipify.org/?format=json"])
+
+        public_ip = {
+            "ip": "undetermined"
+        }
+        try:
+            if self.public_ip is None or time() > self.public_ip_expire:
+                public_ip = json.loads(public_json)
+                self.public_ip = public_ip
+                self.public_ip_expire = time() + 600
+        except:
+            pass
+        data = {
+            "ips": and_(ips),
+            "public": public_ip.get("ip", "undetermined")
+        }
+        self.speak_dialog("ip", data)
+
+    def handle_updtime_intent(self, message):
+        uptime = subprocess.check_output(['uptime', '-p'])
+        data = {
+            'uptime': uptime.strip()
+        }
+        self.speak_dialog("uptime", data)
+
+    def handle_custom_intent(self, message):
+        if not self.diagnostic_script:
+            self.speak_dialog("no.script")
+            return
```

Comment:
What about using `else` instead of `return` to have clearly the 2 flows of this `if`?